tests/test_subtractTask.py

```diff
@@ -0,0 +1,459 @@
+        templateVarianceOrig = template.variance.array[...]
+        for scaleFactor in [1., scaleFactor]:
+            science.variance.array[...] = scienceVarianceOrig/scaleFactor
+            template.variance.array[...] = templateVarianceOrig/scaleFactor
+            scienceVarMean = np.mean(science.variance.array)
+            templateVarMean = np.mean(template.variance.array)
+            config.doSubtractBackground = False
+            for doDecorrelation in [True, False]:
+                for doScaleVariance in [True, False]:
+                    config.doDecorrelation = doDecorrelation
+                    config.doScaleVariance = doScaleVariance
+                    task = subtractImages.AlardLuptonSubtractTask(config=config)
+                    output = task.run(template, science, sources)
+                    if doDecorrelation:
+                        if doScaleVariance:
+                            templateNoise = templateVarMean*scaleFactor
+                            scienceNoise = scienceVarMean*scaleFactor
+                        else:
+                            templateNoise = templateVarMean
+                            scienceNoise = scienceVarMean
+                    else:
+                        if doScaleVariance:
+                            templateNoise = np.mean(output.matchedTemplate.variance.array)*scaleFactor
+                            scienceNoise = scienceVarMean*scaleFactor
+                        else:
+                            templateNoise = np.mean(output.matchedTemplate.variance.array)*scaleFactor
+                            scienceNoise = scienceVarMean
+                    self.assertFloatsAlmostEqual(np.mean(output.difference.variance.array),
+                                                 scienceNoise + templateNoise, rtol=0.1)
+
+    def test_exposure_properties_convolve_template(self):
+        """Check that all necessary exposure metadata is included
+        when the template is convolved.
+        """
+        noiseLevel = 1.
+        seed = 37
+        rng = np.random.RandomState(seed)
+        science, sources = makeTestImage(psfSize=3.0, noiseLevel=noiseLevel, noiseSeed=6)
+        psf = science.psf
+        psfAvgPos = psf.getAveragePosition()
+        psfSize = getPsfFwhm(science.psf)
+        psfImg = psf.computeKernelImage(psfAvgPos)
+        template, _ = makeTestImage(psfSize=2.0, noiseLevel=noiseLevel, noiseSeed=7, templateBorderSize=20)
+
+        # Generate a random aperture correction map
+        apCorrMap = lsst.afw.image.ApCorrMap()
+        for name in ("a", "b", "c"):
+            apCorrMap.set(name, lsst.afw.math.ChebyshevBoundedField(science.getBBox(), rng.randn(3, 3)))
+        science.info.setApCorrMap(apCorrMap)
+
+        config = subtractImages.AlardLuptonSubtractTask.ConfigClass()
+        config.mode = "convolveTemplate"
+        for doDecorrelation in [True, False]:
+            config.doDecorrelation = doDecorrelation
```

Comment:
Can the internal of this loop, and the science one below, become one `_check_something` method? That would also make the tests easier to interpret, because you could then drop the `for` loop and just call it directly with `True` and `False` on separate lines.